### PR TITLE
Add 'connection lost' banner to the UI and display it whenever there is no connection to the server.

### PR DESCRIPTION
**Pull request by marcel-klasse**
Displays a red banner whenever the connection to the matrix server is lost.

- [x] add state within `Matrix.js` to save the current connection state
- [x] display a UI element whenever the connection state is `false`

### CODE REVIEW

File: components/UI/DisplayLostConncetion.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
```
Comment (fnwbr) [inline components/UI/DisplayLostConncetion.js:1]: File name: `Conncetion` or `Connection`?

**Review by fnwbr — APPROVED** [review]

File: components/UI/DisplayLostConncetion.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  display: grid;
+  grid-auto-flow: column;
+  grid-gap: calc(var(--margin)/2);
+  place-items: center;
+  padding: calc(var(--margin)/2) var(--margin);
+  color: white;
+  text-transform: uppercase;
+  background-color: #ff2f4f;
+  border-radius: 4px;
+
```
Comment (fnwbr) [inline components/UI/DisplayLostConncetion.js:22]: @andirueckel Can you please review in terms of UI/UX?

image

File: lib/Matrix.js [hunk 3]
```diff
@@ -172,13 +173,18 @@ function useMatrixProvider(auth) {
...
+        if (newState === 'SYNCING') setIsConnectedToServer(true); // if there is a connection to the server we set our state to true
         if (newState === 'SYNCING' && prevState === 'PREPARED') {
             (async () => {
                 const mDirectEventContent = await matrixClient.getAccountDataFromServer('m.direct');
                 hydrateDirectMessages(mDirectEventContent);
                 setInitialSyncDone(true);
             })();
         }
+        if (newState === 'ERROR' || newState === 'STOPPED') {
+            // if we"ve lost connection to the server we set our state to false
```
Comment (fnwbr) [inline lib/Matrix.js:185]: `we"ve`?

File: pages/_app.js [hunk 4]
```diff
@@ -43,6 +43,7 @@ export default function App({ Component, pageProps }) {
             </Head>
             <AuthContext.Provider value={authData}>
                 <MatrixContext.Provider value={matrixData}>
+                    <DisplayLostConnection />
```
[the remark this answers is not in the document]
Comment (fnwbr) [reply pages/_app.js:46]: Also: `<DisplayLostConnection>` should just be called `<LostConnection>` in my opinion. React components are always about rendering, so it's kind of obvious that the component is responsible for **displaying** _some_ thing...

**Review by fnwbr — CHANGES_REQUESTED** [review]

**Comment by fnwbr** [comment]
I'll defer to @andirueckel for that, but the bottom right red thingy feels kind of out of line with our UI.
How about something bigger at the top, stretching across the whole page?

**Review by andirueckel — APPROVED** [review]

File: components/UI/LostConnection.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  border-radius: 4px;
+
+`;
+const LostConnection = () => {
+    const { t } = useTranslation();
+
+    return (
+        <NoConnectionView>
+            <small>
+                { t('Waiting for server') }
```
Comment (fnwbr) [inline components/UI/LostConnection.js:28]: Missing DE translation for this string

**Review by fnwbr — CHANGES_REQUESTED** [review]
One last thing, otherwise 👍

File: public/locales/de/_defaults.json [hunk 6]
```diff
@@ -2,5 +2,6 @@
     "Password": "Passwort",
     "Select action": "Aktion wählen",
     "What would you like to do?": "Was möchtest Du tun?",
-    "{{error}}, please wait {{time}} seconds.": "{{error}}, bitte warte {{time}} sekunden."
+    "{{error}}, please wait {{time}} seconds.": "{{error}}, bitte warte {{time}} sekunden.",
+    "Connection Lost": "Verbindung unterbrochen"
```
Comment (fnwbr) [inline public/locales/de/_defaults.json:6]: `Connection Lost` vs `Connection lost` -- capital L vs lowercase L...
And while you're at it: Capital `S` in the line above for "Sekunden"

**Review by fnwbr — CHANGES_REQUESTED** [review]

File: components/UI/LostConnection.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  display: grid;
+  grid-auto-flow: column;
+  grid-gap: calc(var(--margin)/2);
+  place-items: center;
+  padding: calc(var(--margin)/2) var(--margin);
+  color: white;
+  text-transform: uppercase;
+  background-color: #ff2f4f;
+  border-radius: 4px;
+
```
Comment (fnwbr) [inline components/UI/LostConnection.js:20]: NAB: Empty line

**Comment by andirueckel** [comment]
> One last thing, otherwise 👍

Merging now. ✌️

**Comment by fnwbr** [comment]
This is still missing the German translation for the `Connection lost` string 😕 
Seems like this happened in your very last merge w/ main @marcel-klasse...

**Comment by marcel-klasse** [comment]
> This is still missing the German translation for the `Connection lost` string 😕 Seems like this happened in your very last merge w/ main @marcel-klasse...

fixed [bb4fa20](https://github.com/medienhaus/medienhaus-spaces/commit/bb4fa20e208436e4572c5a5e71baa5790e1b1597) ✌️

**Pull request merged**